common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -279,6 +287,7 @@ public void initializeApplication(ApplicationInitializationContext context) {
     } catch (Exception e) {
       logger.error("Exception when initializing application {}", appId, e);
     }
+    shuffleMergeManager.registerApplication(appId, context.getUser());
```

Comment:
Are we registering even if we are unable to initialize sasl ?

Reply:
This is a mistake. My next commit will have the fix for this.